extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -406,39 +420,40 @@ protected function sendEmails( array $watchers, WallNotificationEntity $notifica
 				wfRunHooks( 'NotificationGetMailNotificationMessage', [
 					&$notification, &$data, $key, $watcherName, $author_signature, $textNoHtml, $text
 				]);
-				if ( empty( $data ) ) {
-					$data = [
-						'$WATCHER' => $watcherName,
-						'$WIKI' => $notification->data->wikiname,
-						'$PARENT_AUTHOR_NAME' => 	(empty($notification->data->parent_displayname)
-													? ''
-													: $notification->data->parent_displayname),
-						'$AUTHOR_NAME' => $notification->data->msg_author_displayname,
-						'$AUTHOR' => $notification->data->msg_author_username,
-						'$AUTHOR_SIGNATURE' => $author_signature,
-						'$MAIL_SUBJECT' => wfMessage('mail-notification-subject', [
-							'$1' => $notification->data->thread_title,
-							'$2' => $notification->data->wikiname
-						])->text(),
-						'$METATITLE' => $notification->data->thread_title,
-						'$MESSAGE_LINK' =>  $notification->data->url,
-						'$MESSAGE_NO_HTML' =>  $textNoHtml,
-						'$MESSAGE_HTML' =>  $text,
-						'$MSG_KEY_SUBJECT' => $key,
-						'$MSG_KEY_BODY' => 'mail-notification-body',
-						'$MSG_KEY_GREETING' => 'mail-notification-html-greeting',
-					];
-				}
 
-				if ( !( $watcher->getBoolOption('unsubscribed') === true ) ) {
-					$this->sendEmail( $watcher, $data );
-				}
+				$this->sendEmail( $watcher, $data );
 			}
 		}
 
 		return true;
 	}
 
+	protected function canSendToWatcher( User $watcher, $entityKey ) {
+		if ( $watcher->getId() == 0 ) {
+			return false;
+		}
+
+		// Don't send an email to users that unsubscribed their email address
+		if ( $watcher->getBoolOption( 'unsubscribed' ) === true ) {
+			return false;
+		}
+
+		$mode = $watcher->getOption( 'enotifwallthread' );
+		if ( empty( $mode ) ) {
+			return false;
+		}
+
```

Comment:
@kvas-damian FYI, I split this logic up a bit because I accidentally started working on it just before you made your latest changes ;)  This is the only difference we had that I wanted to include...